### PR TITLE
feat: procedures for file table engine

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR implements procedures for the immutable file table engine:
- Adds `CreateImmutableFileTable` for creating a table
- Adds `DropImmutableFileTable` for dropping a table
- ImmutableFileTableEngine now implements the `TableEngineProcedure` trait
- Adds a `common-procedure-test` crate for common test utilities for procedures

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #286

### CODE REVIEW

**Comment by evenyag** [comment]
Please review the related changes to the file table engine. @WenyXu

File: src/file-table-engine/src/engine/procedure/create.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    fn on_prepare(&mut self) -> Result<Status> {
+        let engine_ctx = EngineContext::default();
+        let table_ref = self.data.table_ref();
+        // Safety: Current get_table implementation always returns Ok.
+        if self
+            .engine
+            .get_table(&engine_ctx, &table_ref)
+            .unwrap()
+            .is_some()
```
Comment (MichaelScofield) [inline src/file-table-engine/src/engine/procedure/create.rs:101]:
```suggestion
        if self.engine.table_exists(&engine_ctx, &table_ref)
```

File: src/file-table-engine/src/engine/procedure/create.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    async fn on_create_table(&mut self) -> Result<Status> {
+        let engine_ctx = EngineContext::default();
+        let table_ref = self.data.table_ref();
+        // Safety: Current get_table implementation always returns Ok.
+        if self
+            .engine
+            .get_table(&engine_ctx, &table_ref)
+            .unwrap()
+            .is_some()
```
Comment (MichaelScofield) [inline src/file-table-engine/src/engine/procedure/create.rs:127]:
```suggestion
        if self.engine.table_exists(&engine_ctx, &table_ref)
```

File: src/common/procedure-test/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    async fn procedure_state(&self, _procedure_id: ProcedureId) -> Result<Option<ProcedureState>> {
+        Ok(Some(ProcedureState::Done))
+    }
+}
+
+/// Executes a procedure until it returns [Status::Done].
+///
+/// # Panics
+/// Panics if the `procedure` has subprocedure to execute.
+pub async fn execute_procedure_until_done(procedure: &mut BoxedProcedure) {
```
Comment (MichaelScofield) [inline src/common/procedure-test/src/lib.rs:38]: Can we reuse the `Runner::execute_procedure_in_loop`?

Comment (evenyag) [reply src/common/procedure-test/src/lib.rs:38]: The runner relies on the `ManagerContext`. I want a lightweight helper function to test these simple procedures. We still need such a utility crate so we can add a function to call execute N times in the future. Then the procedure can reach a specific state.

File: src/file-table-engine/src/engine/procedure/drop.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    fn dump(&self) -> Result<String> {
+        let json = serde_json::to_string(&self.data).context(ToJsonSnafu)?;
+        Ok(json)
+    }
+
+    fn lock_key(&self) -> LockKey {
+        // We don't need to support multiple region so we only lock region-0.
+        let table_ref = self.data.table_ref();
+        let key = format!("{table_ref}/region-0");
```
[the remark this answers is not in the document]
Comment (evenyag) [reply src/file-table-engine/src/engine/procedure/drop.rs:61]: The default region is region-0, I keep the same naming pattern as procedures in the mito engine.

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**